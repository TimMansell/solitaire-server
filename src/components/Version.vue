<template>
  <div class="version">
    <a
      href="https://github.com/TimMansell/vue-solitaire"
      target="_blank"
      rel="noopener noreferrer"
      title="View Source Code"
    >
      <span class="version__v">v</span>{{ version }}
    </a>
  </div>
</template>

<script>
import { version } from '../../package.json';

export default {
  name: 'Version',
  data() {
    return {
      version,
    };
  },
};
</script>

<style scoped lang="scss">
.version {
  font-size: var(--font-size-xs);

  @media (min-width: $bp-sm) {
    font-size: var(--font-size-sm);
  }

  &__v {
    font-size: var(--font-size-xs);
  }
}
</style>
