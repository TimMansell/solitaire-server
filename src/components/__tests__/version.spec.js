import { shallowMount } from '@vue/test-utils';
import Version from '@/components/Version.vue';

jest.mock('../../../package.json', () => ({
  version: '1.0.0',
}));

describe('Version.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Version);

    expect(wrapper).toMatchSnapshot();
  });
});
