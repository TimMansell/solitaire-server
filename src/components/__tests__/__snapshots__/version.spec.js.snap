// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Version.vue matches snapshot 1`] = `
<div class="version"><a href="https://github.com/TimMansell/vue-solitaire" target="_blank" rel="noopener noreferrer" title="View Source Code"><span class="version__v">v</span>1.0.0
  </a></div>
`;
